tests/tensorflow/test_tensorflow2_model_export.py

```diff
@@ -134,7 +134,7 @@ def saved_tf_categorical_model(tmpdir):
     ])
     types = collections.OrderedDict((key, type(value[0]))
                                     for key, value in defaults.items())
-    df = pd.read_csv(path, names=types.keys(), dtype=types, na_values="?")
+    df = pd.read_csv(path, names=list(types.keys()), dtype=types, na_values="?")
```

Comment:
Verified this changes fixes the issue in `pandas.io.parsers._validate_names` by:

```python
import collections

import pandas as pd
from pandas.io.parsers import _validate_names

print(pd.__version__)

defaults = collections.OrderedDict([
    ("body-style", [""]),
    ("curb-weight", [0.0]),
    ("highway-mpg", [0.0]),
    ("price", [0.0])
])
types = collections.OrderedDict((key, type(value[0]))
                                for key, value in defaults.items())

_validate_names(list(types.keys()))  # works
_validate_names(types.keys())  # raises an error
```

output:

```
1.1.0
Traceback (most recent call last):
  File "a.py", line 18, in <module>
    _validate_names(types.keys())  # raises an error
  File "/Users/harutakakawamura/.pyenv/versions/miniconda3-4.7.12/envs/mlflow-dev-env/lib/python3.6/site-packages/pandas/io/parsers.py", line 417, in _validate_names
    raise ValueError("Names should be an ordered collection.")
ValueError: Names should be an ordered collection.
```